cmake/templates/setup.bat.in

```diff
@@ -42,10 +42,21 @@ call %_SETUP_TMP%
 del %_SETUP_TMP%
 
 REM source all environment hooks
-for /F "tokens=* delims=;" %%a in ("%_CATKIN_ENVIRONMENT_HOOKS%") do (
-  for %%b in (%%a) do (
-    call "%%b"
-  )
+set _HOOK_COUNT=0
+:hook_loop
+if %_HOOK_COUNT% LSS %_CATKIN_ENVIRONMENT_HOOKS_COUNT% (
+  REM set workspace for environment hook
+  call set CATKIN_ENV_HOOK_WORKSPACE=%%_CATKIN_ENVIRONMENT_HOOKS_%_HOOK_COUNT%_WORKSPACE%%
+  set _CATKIN_ENVIRONMENT_HOOKS_%_HOOK_COUNT%_WORKSPACE=
+
+  REM call environment hook
+  call %%_CATKIN_ENVIRONMENT_HOOKS_%_HOOK_COUNT%%%
+  set _CATKIN_ENVIRONMENT_HOOKS_%_HOOK_COUNT%=
+
+  set CATKIN_ENV_HOOK_WORKSPACE=
+
+  set /a _HOOK_COUNT=%_HOOK_COUNT%+1
+  goto :hook_loop
```

Comment:
Can you please elaborate why this was changed.

Reply:
This is to match what the find_env_hooks() in _setup_util.py template (cmake/templates/_setup_util.py.in) does.  find_env_hooks() will put env hooks to environment variable _CATKIN_ENVIRONMENT_HOOKS_%d and _CATKIN_ENVIRONMENT_HOOKS_%d_WORKSPACE.  This batch file will look for those variables and invoke them